File: manager.ts
import fs from "fs"
import path from "path"
import { ArgumentParser } from "argparse"
import * as toml from "toml"

interface Logo {
  filename: string
  path: string
  size?: number
}

interface Team {
  name: string
  slug: string
  logos: Logo[]
}

interface Category {
  name: string
  slug: string
  teams: Team[]
}

function readTomlFile(filePath: string): any {
  if (!fs.existsSync(filePath)) {
    throw new Error(`Config file missing for ${filePath}`)
  }
  const content = fs.readFileSync(filePath, "utf-8")
  return toml.parse(content)
}

function main(args: { svgPath: string }) {
  const collection: Category[] = []

  const svgFolder = path.resolve(args.svgPath)
  const categories = fs
    .readdirSync(svgFolder, { withFileTypes: true })
    .filter((dirent) => dirent.isDirectory())
    .map((dirent) => path.join(svgFolder, dirent.name))

  for (const cat of categories) {
    const tempTeams: Team[] = []
    const teams = fs
      .readdirSync(cat, { withFileTypes: true })
      .filter((dirent) => dirent.isDirectory())
      .map((dirent) => path.join(cat, dirent.name))

    const categoryConfigPath = path.join(cat, "META.toml")
    const categoryConfig = readTomlFile(categoryConfigPath)

    for (const t of teams) {
      const tempLogos: Logo[] = []
      const svgFiles = fs.readdirSync(t).filter((file) => file.endsWith(".svg"))
      const teamConfigPath = path.join(t, "META.toml")
      const teamConfig = readTomlFile(teamConfigPath)

      for (const f of svgFiles) {
        const relativePath = path.relative(args.svgPath, path.join(t, f))
        const file = Bun.file(path.join(args.svgPath, relativePath))
        tempLogos.push({
          filename: f,
          path: `icons/${relativePath}`,
          size: file.size,
        })
      }

      tempTeams.push({
        name: teamConfig["name"],
        slug: teamConfig["slug"],
        logos: tempLogos,
      })
    }

    collection.push({
      name: categoryConfig["name"],
      slug: categoryConfig["slug"],
      teams: tempTeams,
    })
  }

  console.log(JSON.stringify(collection, null, 2))
}

const parser = new ArgumentParser({
  description: "SVG Manager",
})
parser.add_argument("--svg-path", { required: true })
const args = parser.parse_args()

main({ svgPath: args.svg_path })

File: manager.py
import json
import pathlib
import argparse
import toml
from dataclasses import dataclass, field, asdict


@dataclass
class Logo:
    filename: str
    path: str


@dataclass
class Team:
    name: str
    slug: str
    logos: list[Logo] = field(default_factory=list)


@dataclass
class Category:
    name: str
    slug: str
    teams: list[Team] = field(default_factory=list)


def main(args):
    collection: list[Category] = []

    svg_folder = pathlib.Path(args.svg_path)
    categories = list(svg_folder.iterdir())
    for cat in categories:
        tempTeams: list[Team] = []
        teams = list(filter(lambda x: x.is_dir(), cat.iterdir()))
        category_config = pathlib.Path(cat, "META.toml")
        if not category_config.is_file():
            raise FileNotFoundError(f"Config file missing for {cat}")

        category_config_dict = toml.loads(category_config.read_text())

        for t in teams:
            tempLogos: list[Logo] = []
            svg_files = list(t.glob("*.svg"))
            team_config = pathlib.Path(t, "META.toml")
            if not team_config.is_file():
                raise FileNotFoundError(f"Config file missing for {t}")

            team_config_dict = toml.loads(team_config.read_text())

            for f in svg_files:
                tempLogos.append(Logo(f.name, str(pathlib.Path(*f.parts[1:]))))

            tempTeams.append(
                Team(
                    name=team_config_dict.get("name"),
                    slug=team_config_dict.get("slug"),
                    logos=tempLogos,
                )
            )

        collection.append(
            Category(
                name=category_config_dict.get("name"),
                slug=category_config_dict.get("slug"),
                teams=tempTeams,
            )
        )

    print(json.dumps([asdict(x) for x in collection]))


if __name__ == "__main__":
    parser = argparse.ArgumentParser(
        prog="manager",
    )
    parser.add_argument("--svg-path")
    args = parser.parse_args()

    main(args)
